common/src/main/java/org/ergoplatform/appkit/ErgoTreeTemplate.java

```diff
@@ -64,16 +68,16 @@ public int hashCode() {
      * {@link ErgoTreeTemplate#applyParameters} method.
      * In general, constants of ErgoTree cannot be replaced, but every placeholder can.
      */
-    public int getParameterCount() { return _tree.constants().length(); }
+    public int getParameterCount() {
+        return _tree.constants().length();
+    }
 
     /**
-     * Returns types of all template parameters (placeholders in the ErgoTree).
+     * @param index 0-based
+     * @return value object of paramter
      */
-    public List<ErgoType<?>> getParameterTypes() {
-        Iso<List<Values.Constant<SType>>, IndexedSeq<Values.Constant<SType>>> iso =
-         Iso.JListToIndexedSeq(Iso.identityIso());
-        List<Values.Constant<SType>> ergoValues = iso.from(_tree.constants());
-        return ergoValues.stream().map(v -> Iso.isoErgoTypeToSType().from(v.tpe())).collect(Collectors.toList());
+    public ErgoValue<?> getParameter(int index) {
+        return Iso.isoErgoValueToSValue().from(_tree.constants().apply(index + 1));
     }
```

Comment:
Why getParameterTypes is removed?

Reply:
It returns a SigmaState type that is of not much use in Java alone. The new method `getParameter` returns `ErgoValue` which is the abstraction in appkit and holds the SType if really needed.